20240927-multiple-object-deletion.md

```diff
@@ -0,0 +1,158 @@
+tags: [ "gateway", "satellite", "satellite/metainfo", "uplink" ]
+---
+
+# Multiple Object Deletion
+
+## Essentials
+
+### Header
+
+Date: 2024-09-24
+
+Owner: Jeremy Wharton
+
+Accountable:
+- Edge team
+
+Consulted:
+- Product team
+
+Informed:
+- QA team
+
+### Context
+
+The satellite metainfo RPC service provides an API through which objects stored on the Storj network may be interacted with. While this service exposes an endpoint through which individual objects may be deleted, it lacks an endpoint for deleting multiple objects in a single request. This limitation impacts services that depend on the metainfo RPC service, such as our S3-compatible gateway, by requiring separate requests for each object in a bulk deletion operation.
+
+This approach has several issues:
```

Comment:
Another issue is the Satellite DB overhead. Instead of executing a separate SQL query for each deleted object, those could be batched, reducing the number of SQL queries executed.

I am interested in seeing what the SQL query for deleting multiple objects would look like.

Reply:
Thank you, I added that to the list of issues.

> I am interested in seeing what the SQL query for deleting multiple objects would look like.

Is this a general remark, or are you suggesting that I include such a query in this document?

Reply:
As this is a design document, it is valuable to research how such a query would look (for both Cockroach and Spanner) and then decide whether to make this change with the first milestone or at a later moment.

Reply:
Understood. I pushed a change implementing bulk deletion queries, and I'll update the design doc with them once I have received receive feedback on them. https://review.dev.storj.io/c/storj/storj/+/14974

Reply:
I have updated the design doc with examples of how the deletion queries should be amended. Please let me know whether they are sufficiently detailed or if they should be made more specific to a concrete implementation.

Reply:
(Correspondence was conducted over Slack, so I'll resolve this comment thread.)